driver-core/src/test/unit/com/mongodb/CustomMatchers.groovy

```diff
@@ -55,25 +58,32 @@ class CustomMatchers {
         if (actual.class.name != expected.class.name) {
             return false
         }
-        getFieldNames(actual.class).findAll { !ignoreNames.contains(it) } .collect {
-            if (nominallyTheSame(it)) {
-                return actual."$it".class == expected."$it".class
-            } else if (actual."$it" != expected."$it") {
-                def (a1, e1) = [actual."$it", expected."$it"]
```

Comment:
Since there are now descriptive variable names, got rid of a1, e1 (clarity over brevity)